### ISSUE TITLE
Migrate from `poetry` & `pipx` to `uv` & `uvx`

### ISSUE THREAD

**Issue by janbuchar** [apify/crawlee-python#628]
- `uvx crawlee create` is a bit sexier than `pipx run crawlee create`
- is `uv` as widely available as `pipx`? Windows support?
- any other pitfalls?

**Comment by matt-hayden** [comment]
`uv tool install crawlee` adds a link in `~/.local/bin` to a sandboxed installation.

**Comment by vdusek** [comment]
I am closing #59 as a duplicate since it is closely related to this issue.

TL;DR: Migrate from `poetry` & `pipx` to `uv` & `uvx`.

### Proposed changes

- Use `uv` and `uvx` for:
  - Project bootstrapping
  - Project management
  - Package installation in CI
  - Any other applicable use cases (to be determined)

### Next steps

- Do the research to identify any potential drawbacks or limitations before migrating.

**Comment by vdusek** [comment]
Currently `uv publish` is experimental:

```
warning: `uv publish` is experimental and may change without warning
```

**Comment by vdusek** [comment]
From @janbuchar:

> If we decide to mention uvx as well, the corresponding command would be uvx --with crawlee[cli] crawlee create my-crawler, which is... unsexy. See https://github.com/astral-sh/uv/issues/6296 - the --with flag may become unnecessary in the future.

**Comment by Mantisus** [comment]
> From @janbuchar:
> 
> > If we decide to mention uvx as well, the corresponding command would be uvx --with crawlee[cli] crawlee create my-crawler, which is... unsexy. See [astral-sh/uv#6296](https://github.com/astral-sh/uv/issues/6296) - the --with flag may become unnecessary in the future.

`uvx crawlee create my-crawler` works, `--with crawlee[cli]` is not necessary

**Comment by janbuchar** [comment]
> `uvx crawlee create my-crawler` works, `--with crawlee[cli]` is not necessary

Yep, we're considering adding the `cli` extra in the future so that the generated project doesn't have to depend on `cookiecutter`, `typer` and so on. See #703.

> Windows supports

Great to hear that :slightly_smiling_face:

**Comment by Mantisus** [comment]
> Yep, we're considering adding the `cli` extra in the future so that the generated project doesn't have to depend on `cookiecutter`, `typer` and so on. See #703.

Oh, got it. But it seems to me that for uvx support it is easier to output `cli` dependencies as [script dependencies](https://docs.astral.sh/uv/guides/scripts/#declaring-script-dependencies)

**Comment by janbuchar** [comment]
> > Yep, we're considering adding the `cli` extra in the future so that the generated project doesn't have to depend on `cookiecutter`, `typer` and so on. See #703.
> 
> Oh, got it. But I think it's easier to have `cli` dependencies like [script dependencies](https://docs.astral.sh/uv/guides/scripts/#declaring-script-dependencies) for support uvx.
> 

That looks like something worth checking out, thanks!

**Comment by Mantisus** [comment]
> Currently `uv publish` is experimental:
> 
> ```
> warning: `uv publish` is experimental and may change without warning
> ```

Starting with version 0.6 of `uv publish`, should move out of experimental status

https://github.com/astral-sh/uv/pull/11032

**Comment by vdusek** [comment]
Once we adopt the [PEP 621](https://peps.python.org/pep-0621/) project structure it will break this  https://github.com/apify/workflows/pull/169 and perhaps other things. Do the proper checks, once this is resolved.

**Comment by vdusek** [comment]
`uv` supports [Git dependency sources](https://docs.astral.sh/uv/concepts/projects/dependencies/#git), just like Poetry. This is realy great for development and debugging.

**Comment by Mantisus** [comment]
We may also want to use the setup-uv action in the GitHub Actions flow.

https://docs.astral.sh/uv/guides/integration/github/

**Comment by vdusek** [comment]
`uv` 0.6 released, I believe we can give it a try

**Comment by Pijukatel** [comment]
Seems like it has native support in latest Pycharm as well:
https://blog.jetbrains.com/pycharm/2025/01/pycharm-2024-3-2/

**Comment by vdusek** [comment]
### Solution checklist

- [ ] `apify/workflows`
- [ ] `apify-sdk-python`
- [ ] `apify-client-python`
- [x] `crawlee-python` - https://github.com/apify/crawlee-python/pull/994
- [ ] `actor-templates`
- [ ] `crawlee-python` - templates

**Comment by janbuchar** [comment]
I think we should also change the command on the crawlee docs front page 

![Image](https://github.com/user-attachments/assets/323f7a8e-1d33-4891-af76-2659bba4143e)

**Comment by vdusek** [comment]
Closing, as for the templates we have other issue(s).
